### PR TITLE
Signup form

### PR DESCRIPTION
**Pull request by kazeemerinfolami**
sign up form now working.

### CODE REVIEW

File: fe-local-treasures/styles/signInScreen.js [hunk 1]
```diff
@@ -6,7 +6,7 @@ const styles = StyleSheet.create({
     alignItems: "center",
   },
   controls: {
-    flex: 1,
+    // flex: 1,
```
Comment (ostens) [inline fe-local-treasures/styles/signInScreen.js]: remove comment

File: fe-local-treasures/screens/components/SignUpForm.jsx [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+        placeholder="username"
+        containerStyle={styles.control}
+        value={userInfo.username}
+        onChangeText={(text) => {
+          setUserInfo({ ...userInfo, username: text });
+        }}
+        leftIcon={<Icon name="user" size={16} />}
+      />
+      <Input
+        placeholder="firstname"
```
Comment (ostens) [inline fe-local-treasures/screens/components/SignUpForm.jsx]: placeholder should be First Name

File: fe-local-treasures/screens/components/SignUpForm.jsx [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+        containerStyle={styles.control}
+        value={userInfo.firstname}
+        onChangeText={(text) => {
+          setUserInfo({ ...userInfo, firstname: text });
+        }}
+        leftIcon={<Icon name="user" size={16} />}
+      />
+      <Input
+        placeholder="lastname"
```
Comment (ostens) [inline fe-local-treasures/screens/components/SignUpForm.jsx]: Placeholder should be Last Name

File: fe-local-treasures/screens/components/SignUpForm.jsx [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+  uploadProgress,
+  progressBar,
+  selectedImage,
+  pickImage,
+  signUp,
+}) {
+  return (
+    <View style={styles.controls}>
+      <Input
+        placeholder="username"
```
Comment (ostens) [inline fe-local-treasures/screens/components/SignUpForm.jsx]: Placeholder should be Username

File: fe-local-treasures/screens/SignUpScreen.jsx [hunk 5]
```diff
@@ -1,155 +1,155 @@
...
+              break;
+          }
+        },
+        (error) => {
+          switch (error.code) {
+            case "storage/unauthorized":
+              break;
+            case "storage/canceled":
+              break;
+            // ...
```
Comment (ostens) [inline fe-local-treasures/screens/SignUpScreen.jsx]: remove this I think

File: fe-local-treasures/screens/SignUpScreen.jsx [hunk 6]
```diff
@@ -1,155 +1,155 @@
...
+        };
+        xhr.onerror = function () {
+          reject(new TypeError("Network request failed"));
+        };
+        xhr.responseType = "blob";
+        xhr.open("GET", selectedImage, true);
+        xhr.send(null);
+      });
+
+      /** @type {any} */
```
Comment (ostens) [inline fe-local-treasures/screens/SignUpScreen.jsx]: remove this I think?

File: fe-local-treasures/utils/firebase/imageUpload.js [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+    quality: 1,
+  });
+  if (!result.canceled) {
+    setSelectedImage(result.assets[0].uri);
+  } else {
+    alert("You did not select any image.");
+  }
+};
+
+export const uploadImage = async () => {
```
Comment (ostens) [inline fe-local-treasures/utils/firebase/imageUpload.js]: Not sure if this is somewhat duplicating the code from line 50 on the sign up screen

File: fe-local-treasures/screens/SignUpScreen.jsx [hunk 8]
```diff
@@ -1,155 +1,155 @@
...
     } else {
       alert("You did not select any image.");
     }
   };
 
-  function signUp() {
+  useEffect(() => {
+    const uploadImage = async () => {
+      const blobImage = await new Promise((resolve, reject) => {
+        const xhr = new XMLHttpRequest();
```
Comment (ostens) [inline fe-local-treasures/screens/SignUpScreen.jsx:52]: would be nice to explain to the team how this section works :)

File: fe-local-treasures/screens/SignUpScreen.jsx (hunk 6 above)
Comment (kazeemerinfolami) [reply fe-local-treasures/screens/SignUpScreen.jsx]: This is part of the initial setup for the image upload

**Pull request merged**